pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -652,3 +671,59 @@ func (nm *NodeManager) getNodeNameByUUID(UUID string) string {
 	}
 	return ""
 }
+
+func extractGuestInfo(extraConfig []types.BaseOptionValue) (string, string) {
+	var guestInfo, encoding string
+	for _, option := range extraConfig {
+		value := option.GetOptionValue()
+		switch value.Key {
+		case "guestinfo.metadata":
+			guestInfo, _ = value.Value.(string)
+		case "guestinfo.metadata.encoding":
+			encoding, _ = value.Value.(string)
+		}
+	}
+	return guestInfo, encoding
+}
+
+// sortStaticallyConfiguredAddressesFirst prefers addresses that are from the
+// guestInfo but only if they are on a NIC already. It preserves the order in which
+// the addresses appear in the guestInfo. For addresses not found in the guestInfo,
+// it preserves the order in which they appear in nonlocalhostIPs.
+func sortStaticallyConfiguredAddressesFirst(extraConfig []types.BaseOptionValue, nonLocalhostIPs []*ipAddrNetworkName) ([]*ipAddrNetworkName, error) {
+	guestInfo, encoding := extractGuestInfo(extraConfig)
+
+	if guestInfo != "" && encoding == "base64" {
+		value, err := base64.StdEncoding.DecodeString(guestInfo)
+		if err != nil {
+			return nil, err
+		}
+
+		guestInfo := &cloudInitConfig{}
+		err = yaml.Unmarshal(value, guestInfo)
+		if err != nil {
+			return nil, err
+		}
+
+		// Map of guestInfo IP -> index that describes the order they appear in the guestInfo
+		guestInfoAddresses := make(map[string]int)
+		for _, eth := range guestInfo.Network.Ethernets {
```

Comment:
Could we check whether Network is also not nil

Reply:
Network is not a pointer so it shouldn't be nil in this case.

Reply:
Oh sorry you are right